samples/nrf9160/aws_iot/src/main.c

```diff
@@ -176,6 +176,33 @@ static void shadow_update_version_work_fn(struct k_work *work)
 	}
 }
 
+static void print_received_data(const char *buf, const char *topic,
+				size_t topic_len)
+{
+	char *str = NULL;
+	cJSON *root_obj = NULL;
+
+	root_obj = cJSON_Parse(buf);
+	if (root_obj == NULL) {
+		printk("cJSON Parse failure");
+		return;
+	}
+
+	str = cJSON_Print(root_obj);
+	if (str == NULL) {
+		printk("Failed to print JSON object");
+		goto clean_exit;
+	}
+
+	printf("Data received from AWS IoT console:\nTopic: %.*s\nMessage: %s\n",
+	       topic_len, topic, str);
+
+	k_free(str);
```

Comment:
nit; There's a `cJSON_FreeString` helper in the `cJSON_os.h` header, would be good to use it instead of direct `k_free()` call.

Reply:
Changed for both occurrences in main.c